### PR TITLE
feat: Banner

### PR DESCRIPTION
**Pull request by PuruVJ**

### CODE REVIEW

File: packages/site-kit/src/lib/components/Shell.svelte [hunk 1]
```diff
@@ -60,15 +56,13 @@ The main shell of the application. It provides a slot for the top navigation, th
 
 <div class="modal-overlay" class:visible={$overlay_open} aria-hidden="true" />
 
-<main id="main" bind:this={main_el} style:--sk-banner-bottom-height={banner_bottom_height}>
```
Comment (dummdidumm) [inline packages/site-kit/src/lib/components/Shell.svelte:63]: why was this part removed? AFAIK it's needed, else the main content will go up to the bottom and might get cut off by the banner that is also there - worse, it might result in a needless double scrollbar.

Comment (PuruVJ) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: I took care of that in Banner.svelte. Now it works even better than before

Comment (dummdidumm) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: I'm not too sure about that. This way, it would flicker after server side rendering when the banner is present. On the other hand, if we did render it in SSR with the height already, it would also flicker when someone closed the banner and it's persisted and now it shortly reappears on SSR. Mhhhhhhhhhm there's no perfect answere here, sad.

Comment (PuruVJ) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: Banner by default is not rendered. Only flicker on loading is when it appears. While disappearing, there's no flicker. And initially, height is 0, so its fully hidden, so not jarring either

Comment (dummdidumm) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: The flicker will be that for a very short time there will be no banner, and then it appears, possible creating a scrollbar, or otherwise shifting around content a bit. That's my concern here. Makes me wonder if the persistent store is worth it.

Comment (PuruVJ) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: The shifting will be of the REPL height, and javascript typescript toggle. Both unnoticeable I'd say. 

> Makes me wonder if the persistent store is worth it.

I'd say it is. If I dismiss it once, I probably don't wanna see it on every reload. Flickering-wise, getting rid of persistent will fix that tiny issue. Keeping the user's preference-wise, not as good. If it's still important to present it to the user, I could use session storage. I believe this library allows for session storage now, so having 2 day timer for banner store wouldn't be too bad

Comment (benmccann) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: Should we use a cookie? Then I believe we could correctly determine the value on the client and server. We'd have to give each banner a unique key so that if you closed a banner previously you're not stopped from seeing all future banners

Comment (PuruVJ) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: Wouldn't work I believe, the pages are prerendered, so only repl page can take advantage of having a cookie

Comment (benmccann) [reply packages/site-kit/src/lib/components/Shell.svelte:63]: Ah, right. Anyway, I do think a slight flicker will be less annoying to folks than not having the option to dismiss the banner
